### ISSUE TITLE
data loading error: _pickle.UnpicklingError: pickle data was truncated

### ISSUE THREAD

**Issue by wenliangdai** [yaohungt/Multimodal-Transformer#22]
Hi, I found a problem when loading the data.

I follow the instructions of this project, and when I run `python main.py`, an error happened when loading the pickle file

```
Start loading the data....
  - Creating new train data
Traceback (most recent call last):
  File "main.py", line 118, in <module>
    train_data = get_data(args, dataset, 'train')
  File "/Users/wenliangdai/Documents/Codes/Multimodal-Transformer/src/utils.py", line 11, in get_data
    data = Multimodal_Datasets(args.data_path, dataset, split, args.aligned)
  File "/Users/wenliangdai/Documents/Codes/Multimodal-Transformer/src/dataset.py", line 23, in __init__
    dataset = pickle.load(open(dataset_path, 'rb'))
_pickle.UnpicklingError: pickle data was truncated
```

If I add `--dataset=mosi` or `--dataset=iemocap`, there is no such error. Seems this only happens with the mosei_senti dataset.

Could you please tell me how to fix this? If that's a problem of the data file is broken, could you please provide new data files?

Thanks a lot!

**Comment by yaohungt** [comment]
Hi, I never encounter this error in our dataset. 

If it is pickle error, I'm guessing there's something wrong when you download the data. Do you mind to re-download the data again?

**Comment by wenliangdai** [comment]
@yaohungt 

Hi, Thanks for your reply. 

I tried to download the data again (from the dropbox link that you provided), the same error still exists. Only mosei_senti dataset has this problem.

If you don't encounter, could you please try to delete the cached data and use the data on Dropbox to try again? 

Thank you a lot!

**Comment by yaohungt** [comment]
Unfortunately, there is no problem on my side.

**Comment by mnc537** [comment]
Hi! 

I found the same problem. I re-downloaded the data (.zip) several times and the error persists. I don't get any error when I unzip the downloaded data file either. 

Does anyone know what could it be? 

Thanks

**Comment by wenliangdai** [comment]
@mnc537 Hi, I solved this problem by using the original data to re-generate the saved torch Dataset again, rather than using the one from the dropbox link, I think that one is somehow broken

**Comment by yaohungt** [comment]
No, it's not borken. Change the encoding specification in the pickle package, to bytes or latin1

**Comment by wenliangdai** [comment]
I see, so that file is saved by Python2?

**Comment by yaohungt** [comment]
It might be, I checked it when you open this issue, and nothing strange on my side. It also might be the case that I store it with Python3 and you can't directly open it with Python 2. In short, I think it's a python-version issue.

**Comment by wenliangdai** [comment]
Yes, I agree, but I think it's the other way around. The file is saved by python 2 and I try to load with python 3. 
Something like this problem https://stackoverflow.com/questions/11305790/pickle-incompatibility-of-numpy-arrays-between-python-2-and-3

**Comment by mnc537** [comment]
I tried to open it with Python2 and it doesn't work. I get this error: `ValueError: unsupported pickle protocol: 4`. From the [documentation](https://docs.python.org/3/library/pickle.html#:~:text=Protocol%20version%204%20was%20added,improvements%20brought%20by%20protocol%204.): 
> Pickle protocol version 4 was added in Python 3.4

So the data was store in Python3.

Also, I changed the encoding specification to bytes and latin1 with Python3 and it doesn't work either. I think that something went wrong when the data was store since it's only one pkl that have this error. I can open the other ones without any issue. 

@wenliangdai Great! I'll do the same.

Thanks a lot!
